api/server/router/image/image_routes.go

```diff
@@ -215,15 +215,15 @@ func (ir *imageRouter) postImagesPush(ctx context.Context, w http.ResponseWriter
 		if formPlatform := r.Form.Get("platform"); formPlatform != "" {
 			p, err := httputils.DecodePlatform(formPlatform)
 			if err != nil {
-				return err
+				return errdefs.InvalidParameter(err)
```

Comment:
DecodePlatform above already returns a typed error, so this should not be needed https://github.com/moby/moby/blob/980d8e5d54d8df31ba0e75e1d9dd474d9d91308a/api/server/httputils/form.go#L119